xarray/core/dataset.py

```diff
@@ -2182,14 +2182,14 @@ def isel(
             If DataArrays are passed as indexers, xarray-style indexing will be
             carried out. See :ref:`indexing` for the details.
             One of indexers or indexers_kwargs must be provided.
-        drop : bool, optional
+        drop : bool, default False
```

Comment:
Weirdly, this doesn't actually get used in `_isel_fancy`...

Reply:
https://github.com/pydata/xarray/pull/6580

Reply:
Isn't that destroying the purpose of this PR?

Reply:
It gets used just below! Atm `isel_fancy` doesn't do anything with `drop` — unless I'm missing something (very possible)

Reply:
`drop` gets used here: https://github.com/pydata/xarray/pull/6580/files#diff-763e3002fd954d544b05858d8d138b828b66b6a2a0ae3cd58d2040a652f14638R2236